examples/src/test/scala/GithubExample.scala

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright 2016-2018 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+import cats._
+import cats.data.NonEmptyList
+import cats.effect._
+import cats.instances.list._
+import cats.syntax.all._
+import cats.temp.par._
+
+import io.circe._
+import io.circe.generic.semiauto._
+
+import org.http4s._
+import org.http4s.headers._
+import org.http4s.util.CaseInsensitiveString
+import org.http4s.circe._
+import org.http4s.client._
+import org.http4s.client.dsl._
+import org.http4s.client.blaze._
+import org.scalatest.{Matchers, WordSpec}
+
+import fetch.{DataSource, Env, Fetch}
+
+class GithubExample extends WordSpec with Matchers {
+  implicit val executionContext = ExecutionContext.Implicits.global
+
+  val ACCESS_TOKEN: String = sys.env("ORG_GITHUB_TOKEN")
+
+  implicit val t: Timer[IO]         = IO.timer(executionContext)
+  implicit val cs: ContextShift[IO] = IO.contextShift(executionContext)
+
+  // http4s client which is used by the datasources
+
+  def client[F[_]: ConcurrentEffect] =
+    Http1Client[F](BlazeClientConfig.defaultConfig)
+
+  // Github HTTP api
+
+  val GITHUB: Uri = Uri.unsafeFromString("https://api.github.com")
+
+  val REL_NEXT = "rel=\"next\"".r
+
+  def hasNext[F[_]: ConcurrentEffect](res: Response[F]): Boolean =
```

Comment:
This function is used only inside `private` functions, can we define it as `private` too?